lib/compress/zstdmt_compress.c

```diff
@@ -714,7 +768,7 @@ size_t ZSTDMT_initCStream_internal(ZSTDMT_CCtx* zcs,
         DEBUGLOG(4,"cdictLocal: %08X", (U32)(size_t)zcs->cdictLocal);
         ZSTD_freeCDict(zcs->cdictLocal);
         zcs->cdictLocal = ZSTD_createCDict_advanced(dict, dictSize,
-                                                    0 /* byRef */, ZSTD_dm_auto,   /* note : a loadPrefix becomes an internal CDict */
+                                                    ZSTD_dlm_byCopy, ZSTD_dm_rawContent, /* note : a loadPrefix becomes an internal CDict */
```

Comment:
Why are we forcing `rawContent` now?

Reply:
This might be related to the discussion @stellamplau and I had yesterday : 
https://github.com/facebook/zstd/pull/810#discussion_r135933032
In short, `ZSTD_refPrefix()` now assumes `ZSTD_dm_rawContent`.

Still I don't think it's the right place to enforce this mode.
It should be enforced from the calling side, aka `ZSTD_compress_generic()`, which means `ZSTDMT_initCStream_internal()` should have another argument, `dictMode` (only useful when `dict` is defined)

The reason is, `ZSTDMT_initCStream_internal()` is also accessible through another entry point, `ZSTDMT_initCStream_advanced()`, which makes it possible to select a `dict`, but for which the implied mode is `ZSTD_dm_auto`.

Passing the `dictMode` is the only way to make the distinction between these 2 entry points.

Reply:
Yeah I was just about to comment that forcing `rawContent` here was a mistake and the function should probably contain a `dictMode` parameter. I neglected the `ZSTDMT_initCStream_advanced` caller.

Reply:
Lets add a test for `ZSTDMT_initCStream_advanced()` with a dictionary, since it looks like we don't have one already.